core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -3150,6 +3143,26 @@ object Stream extends StreamLowPriority {
   /** Creates a pure stream that emits the supplied values. To convert to an effectful stream, use `covary`. */
   def apply[F[x] >: Pure[x], O](os: O*): Stream[F, O] = emits(os)
 
+  def async[F[x] >: Pure[x], O](maxWaiting: Int)(f: Enqueue[F, O] => Stream[F, Unit])(implicit
```

Comment:
I think if we want these to reduce the learning curve, we're going to need to add some javadoc to them, and some usage examples somewhere

Reply:
Yep, happy to add those if/when the API is agreed